tests/e2e/test_small_file_workload.py

```diff
@@ -0,0 +1,75 @@
+"""
+Test to exercise Small File Workload
+"""
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.ocs import OCS
+from ocs_ci.utility import templating
+from ocs_ci.ocs.utils import get_pod_name
+from ocs_ci.ocs.container import ContainerStor
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import E2ETest, tier1
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def ripsaw(request):
+    # Create Ceph Block Pool
+    sc = ContainerStor(sc_name='ceph-backed', cleanup=True)
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+    sc.create()
+
+    def teardown():
+        ripsaw.cleanup()
+        sc.delete()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@tier1
+class TestSmallFileWorkload(E2ETest):
+    """
+    Deploy Ripsaw operator and run SmallFile workload
+    """
+    def test_smallfile_workload(self, ripsaw):
+        """
+        Run SmallFile Workload
+        """
+        log.info("Apply Operator CRD")
+        ripsaw.apply_crd('resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml')
+
+        log.info("Running SmallFile bench")
+        sf_data = templating.load_yaml_to_dict(constants.SMALLFILE_BENCHMARK_YAML)
+        sf_obj = OCS(**sf_data)
+        sf_obj.create()
+        # wait for benchmark pods to get created - takes a while
+        time.sleep(20)
+        small_file_client_pod = get_pod_name('smallfile-client', 'my-ripsaw')
+        bench_pod = OCP(kind='pod', namespace='my-ripsaw')
+        log.info("Waiting for SmallFile benchmark to Run")
+        bench_pod.wait_for_resource(
+            condition='Running',
+            resource_name=small_file_client_pod[0],
+            sleep=30,
+            timeout=600
+        )
+        start_time = time.time()
+        timeout = 600
+        while True:
+            logs = bench_pod.exec_oc_cmd(
+                f'logs {small_file_client_pod[0]}',
+                out_yaml_format=False
+            )
+            if "RUN STATUS DONE" in logs:
+                log.info("SmallFile Benchmark Completed Successfully")
+                return True
+
+            if timeout < (time.time() - start_time):
+                raise TimeoutError(f"Timed out waiting for benchmark to complete")
+            time.sleep(30)
```

Comment:
same regarding sleep

Reply:
The sleep here is to make sure we are not repeatedly getting the logs, the loop is kind of sample loop (wait -check for logs - assert on success or keep polling)